lib/oelite/parse/oeparse.py

```diff
@@ -582,13 +582,13 @@ def inherit(self, filename, p):
             return
         if not os.path.isabs(filename) and not filename.endswith(".oeclass"):
             filename = os.path.join("classes", "%s.oeclass"%(filename))
-        if not "__inherits" in self.meta:
-            self.meta["__inherits"] = [filename]
-        else:
-            __inherits = self.meta["__inherits"]
+        __inherits = self.meta.get("__inherits")
+        if __inherits:
```

Comment:
What about
```python
__inherits = self.meta.setdefault("__inherits", [])
if filename not in __inherits:
    __inherits.append(filename)
```
which also takes care of the else clause below.  The `[filename]` is processed in all cases though, causing an overhead...